src/ssh/azext_ssh/ssh_utils.py

```diff
@@ -35,43 +35,61 @@ def start_ssh_connection(op_info, delete_keys, delete_cert):
         # When connecting to a local user on a host with a banner, output gets messed up if stderr redirected.
         # If user expects logs to be printed, do not redirect logs. In some ocasions output gets messed up.
         redirect_stderr = set(['-v', '-vv', '-vvv']).isdisjoint(ssh_arg_list) and \
-            (delete_cert or op_info.delete_credentials)
-
+            (delete_cert or op_info.delete_credentials or op_info.new_service_config)
```

Comment:
I'm not sure if we want to redirect the logs every time there is a new service config, but that is an interesting idea. I'll give some thought on that.

Reply:
Updated this. I think since we check for exit code 255 check if logs were not redirected, it's probably fine to not automatically redirect the logs.